### PR TITLE
✨ feat: add collaborator management for partner users

### PR DESCRIPTION
**Pull request by BenoitSerrano**
**Problem**
Users of the Partner Portal cannot invite collaborators to manage their Service Providers.

**Proposal**
Add a `collaborators` field to the database and provide the ability to manage it through the admin interface.

### CODE REVIEW

**Comment by BenoitSerrano** [comment]
[Ajouter un champ "collaborators" éditable dans l'admin](https://trello.com/c/HqtuPWYi/1463-ajouter-un-champ-collaborators-%C3%A9ditable-dans-ladmin)

**Review by rdubigny — APPROVED** [review]

File: admin/src/service-provider/dto/service-provider-input.dto.spec.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ describe("Service Provider Input (Data Transfer Object)", () => {
     active: "true",
     type: "private",
     scopes: ["given_name"],
+    collaborators: [],
```
Comment (rdubigny) [inline admin/src/service-provider/dto/service-provider-input.dto.spec.ts:15]: Thoughts: I would have liked to see more tests for this field validation, such as collaborators with empty string, `null`, or `undefined` values.
